packages/block-editor/src/components/block-list/block-list-item.native.js

```diff
@@ -89,6 +93,52 @@ export class BlockListItem extends Component {
 		];
 	}
 
+	componentDidUpdate() {
+		this.scrollToSelectedBlockIfItsNotVisible();
+	}
+
+	scrollToSelectedBlockIfItsNotVisible() {
+		const { isSelected } = this.props;
+
+		if ( isSelected ) {
+			const isIOS = Platform.OS === 'ios';
+
+			// this adjustment factor is utilized to create an invisible bounds at the top and bottom of the screen where once the View enters within
+			//any of these bounds then the scrolling action is triggered.
+			const scrollAdjustmentY = isIOS ? 200 : 100;
```

Comment:
I am always concerned about hardcoding these types of values because I am not sure if they will support every screen size. I haven't done a thorough test on multiple screen sizes as yet as I wanted to get some opinions on the proposed solution here.

Reply:
I ended up going for a more dynamic solution using the window's height. [Done here ](https://github.com/WordPress/gutenberg/pull/26596/commits/29a0ed129210e962bc43643543867a950fed2c80#diff-d2ed3f81be347a76e4b1ce3e9ddf8723b0bb8a532207ab5a46e9680912373648R118-R121)